src/lib/builtins/string.rs

```diff
@@ -728,6 +728,41 @@ pub fn match_all(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultV
     regexp_match_all(&re, ctx.value_to_rust_string(this))
 }
 
+/// Return a String where the matches of a pattern in the original string is replaced with a given
+/// replacement.
+/// <https://tc39.es/ecma262/#sec-string.prototype.replace>
+pub fn replace(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
+    // First we get it the actual string a private field stored on the object only the engine has access to.
+    // Then we convert it into a Rust String by wrapping it in from_value
+    let this_str: &str = &ctx.value_to_rust_string(this);
+    // We then do the same thing for the other args
+    // TODO: what if there are missing arguments?
+    let pattern = get_argument(args, 0);
+    if undefined() == pattern {
```

Comment:
If I am not mistaken, there was an `is_undefined` method.
So you can try `pattern.is_undefined()` instead.

Reply:
Good catch!